### PR TITLE
Reduced SoPointer usage in internal API

### PR DESCRIPTION
**Pull request by apankratovantonp**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: inference-engine/tests/unit/inference_engine/cpp_interfaces/ie_memory_state_internal_test.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ class InferRequestVariableStateTests : public ::testing::Test {
     shared_ptr<MockIVariableStateInternal> mockVariableStateInternal;
     MockIInferencePlugin*                           mockIPlugin;
     InferencePlugin                                 plugin;
-    SoExecutableNetworkInternal                     net;
+    details::SOPointer<IExecutableNetworkInternal>  net;
```
Comment (ilya-lavrenov) [inline inference-engine/tests/unit/inference_engine/cpp_interfaces/ie_memory_state_internal_test.cpp:30]: can we remove `SOPointer` from here as well?

File: inference-engine/tests/unit/inference_engine/ie_executable_network_test.cpp [hunk 2]
```diff
@@ -38,7 +38,7 @@ using testing::SetArgReferee;
 class ExecutableNetworkTests : public ::testing::Test {
 protected:
     std::shared_ptr<MockIExecutableNetworkInternal> mockIExeNet;
-    InferenceEngine::SoExecutableNetworkInternal    exeNetwork;
+    details::SOPointer<IExecutableNetworkInternal>  exeNetwork;
```
Comment (ilya-lavrenov) [inline inference-engine/tests/unit/inference_engine/ie_executable_network_test.cpp:41]: and here

File: inference-engine/tests/unit/inference_engine/cpp_interfaces/ie_memory_state_internal_test.cpp (hunk 1 above)
Comment (apankratovantonp) [reply inference-engine/tests/unit/inference_engine/cpp_interfaces/ie_memory_state_internal_test.cpp]: done

File: inference-engine/tests/unit/inference_engine/ie_executable_network_test.cpp (hunk 2 above)
Comment (apankratovantonp) [reply inference-engine/tests/unit/inference_engine/ie_executable_network_test.cpp]: done

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**